File: CS/FederationDataSourceExample/Form1.cs
using DevExpress.DataAccess.ConnectionParameters;
using DevExpress.DataAccess.DataFederation;
using DevExpress.DataAccess.Excel;
using DevExpress.DataAccess.Sql;
using DevExpress.XtraEditors;
using System;
using System.Reflection;
using System.Windows.Forms;

namespace FederationDataSourceExample
{
    public partial class Form1 : DevExpress.XtraEditors.XtraForm
    {
        public Form1()
        {
            InitializeComponent();
            
            foreach (var method in typeof(FederationDataSourceHelper).GetMethods(BindingFlags.DeclaredOnly | BindingFlags.Static |
                BindingFlags.Public))
            {
                if (!method.IsSpecialName)
                    comboBoxEdit1.Properties.Items.Add(method.Name);
            }
            comboBoxEdit1.Properties.TextEditStyle = DevExpress.XtraEditors.Controls.TextEditStyles.DisableTextEditor;
            comboBoxEdit1.Properties.SelectedValueChanged += Properties_SelectedValueChanged;

        }

        private void Properties_SelectedValueChanged(object sender, EventArgs e)
        {
            gridView1.BeginUpdate();
            gridView1.Columns.Clear();
            gridControl1.DataSource = null;
            ComboBoxEdit editor = sender as ComboBoxEdit;
            typeof(FederationDataSourceHelper).InvokeMember(editor.SelectedItem.ToString(), BindingFlags.InvokeMethod | BindingFlags.Public |
                BindingFlags.Static, null, null, new object[] { });
            gridControl1.DataSource = FederationDataSourceHelper.DataSource;
            gridView1.EndUpdate();
            gridView1.ExpandMasterRow(0);
        }
    }
}

File: VB/FederationDataSourceExample/Form1.vb
Imports DevExpress.DataAccess.ConnectionParameters
Imports DevExpress.DataAccess.DataFederation
Imports DevExpress.DataAccess.Excel
Imports DevExpress.DataAccess.Sql
Imports DevExpress.XtraEditors
Imports System
Imports System.Reflection
Imports System.Windows.Forms

Namespace FederationDataSourceExample
	Partial Public Class Form1
		Inherits DevExpress.XtraEditors.XtraForm

		Public Sub New()
			InitializeComponent()

			For Each method In GetType(FederationDataSourceHelper).GetMethods(BindingFlags.DeclaredOnly Or BindingFlags.Static Or BindingFlags.Public)
				If Not method.IsSpecialName Then
					comboBoxEdit1.Properties.Items.Add(method.Name)
				End If
			Next method
			comboBoxEdit1.Properties.TextEditStyle = DevExpress.XtraEditors.Controls.TextEditStyles.DisableTextEditor
			AddHandler comboBoxEdit1.Properties.SelectedValueChanged, AddressOf Properties_SelectedValueChanged

		End Sub

		Private Sub Properties_SelectedValueChanged(ByVal sender As Object, ByVal e As EventArgs)
			gridView1.BeginUpdate()
			gridView1.Columns.Clear()
			gridControl1.DataSource = Nothing
			Dim editor As ComboBoxEdit = TryCast(sender, ComboBoxEdit)
			GetType(FederationDataSourceHelper).InvokeMember(editor.SelectedItem.ToString(), BindingFlags.InvokeMethod Or BindingFlags.Public Or BindingFlags.Static, Nothing, Nothing, New Object() { })
			gridControl1.DataSource = FederationDataSourceHelper.DataSource
			gridView1.EndUpdate()
			gridView1.ExpandMasterRow(0)
		End Sub
	End Class
End Namespace
